### PR TITLE
Update invoice status on changes to distribution code

### PR DESCRIPTION
**Pull request by sumesh-aot**
*Issue #:*
https://github.com/bcgov/entity/issues/4373

*Description of changes:*
Update invoice status on changes to distribution code

*Checklist:*
I confirm that below checklist items are addressed in this pull request; (check the boxes applicable)
- [ ] No lint errors
- [ ] No test case failures and proper coverage for all modules/classes
- [ ] Updated the deployment configs for new environment variable(s)
- [ ] Updated the postman collection in entity repository (https://github.com/bcgov/entity/tree/master/api-e2e/postman) for e2e tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-pay license (Apache 2.0).

### CODE REVIEW

File: pay-api/src/pay_api/services/payment_transaction.py [hunk 1]
```diff
@@ -310,7 +310,7 @@ def update_transaction(payment_identifier: int, transaction_id: uuid,  # pylint:
         except ServiceUnavailableException as exc:
             txn_reason_code = exc.status
             receipt_details = None
-
+        print('0000000', receipt_details)
```
Comment (pwei1018) [inline pay-api/src/pay_api/services/payment_transaction.py:313]: need print?

File: pay-api/src/pay_api/models/invoice.py [hunk 2]
```diff
@@ -76,6 +76,20 @@ def find_by_id_and_payment_id(cls, identifier: int, pay_id: int):
...
+    @classmethod
+    def update_invoices_for_revenue_updates(cls, fee_distribution_id: int):
+        """Find and update all invoices using the distribution id."""
+        query = db.session.query(Invoice) \
+            .join(PaymentLineItem, PaymentLineItem.invoice_id == Invoice.id) \
+            .filter(PaymentLineItem.fee_distribution_id == fee_distribution_id)
+
+        invoices: [Invoice] = query.all()
+        for invoice in invoices:
+            if invoice.invoice_status_code == InvoiceStatus.PAID.value:
```
Comment (saravanpa-aot) [inline pay-api/src/pay_api/models/invoice.py:88]: shouldn't we do for Unpaid invoices as well? not sure if thats gonna happen for direct pay..but was thinking of user going to bambora screen and closing screen kind of edge cases... 
also should we have a date check as well? like may be 1 week or 1 month time ; will records before that have to be rectified in paybc?

File: pay-api/src/pay_api/services/payment_transaction.py (hunk 1 above)
Comment (sumesh-aot) [reply pay-api/src/pay_api/services/payment_transaction.py:313]: Will remove this in next PR

File: pay-api/src/pay_api/models/invoice.py (hunk 2 above)
Comment (sumesh-aot) [reply pay-api/src/pay_api/models/invoice.py:88]: Unpaid invoices we can think about it. Didn't want to handle bunch of 404 from PayBC. The job to update stale payment will be 30 minutes, so we shouldn't see lot of those issues. 
 
The job will check the GL status before updating the revenue account. This feature, we are considering only for any error in GL code. If the use case is for new GL code, we should be creating new GL code and map the filing types to GL code.

Comment (saravanpa-aot) [reply pay-api/src/pay_api/models/invoice.py:88]: yup.. thats good ..

**Pull request merged**